File: 02_open_closed/typescript/good/type_question/multple_choice_question.ts
import { PrintQuestion } from "./print_question.ts";

export class MultpleChoiceQuestion implements PrintQuestion {
  description: string;
  options: Array<string>;

  constructor(description: string, options: Array<string>) {
    this.description = description
    this.options = options
  }
  printQuestionChoices(): void {
    this.options.forEach((option, index) => {
      console.log(`${index + 1}. ${option}`)
    })
  }
}

File: 02_open_closed/dart/good/type_question/multple_choice_question.dart
import 'print_question.dart';

class MultpleChoiceQuestion extends PrintQuestion {
  MultpleChoiceQuestion(super.description, this.options);

  final List<String> options;

  void printQuestionChoices() {
    this.options.toList().asMap().forEach((int index, String option) {
      print('${index + 1}. $option');
    });
  }
}
